webpack/InsightsCloudSync/Components/InsightsSettings/InsightsSettingsReducer.js

```diff
@@ -0,0 +1,28 @@
+import Immutable from 'seamless-immutable';
+import {
+  INSIGHTS_SYNC_SETTING_SET,
+  INSIGHTS_SYNC_SETTING_SET_ERROR,
+  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+} from './InsightsSettingsConstants';
+
+const initialState = Immutable({
+  insightsSyncEnabled: false,
+});
+
+export default (state = initialState, action) => {
+  const { payload: { settings } = {} } = action;
+
+  switch (action.type) {
+    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS:
+    case INSIGHTS_SYNC_SETTINGS_GET_FAILURE:
+    case INSIGHTS_SYNC_SETTING_SET:
+    case INSIGHTS_SYNC_SETTING_SET_ERROR:
```

Comment:
as the action passes the [error message in payload](https://github.com/theforeman/foreman_rh_cloud/pull/277/files#diff-ad0bf04f6b972ee2294b94466357f7a8R28) we should handle the error a bit differently.

Reply:
Error json looks like this:
``` json
"settings": {
  "error": "error message",
},
```
I think it should be destructed correctly by this statement.

Reply:
Nice ! Didn't spot the action payload structure :)